feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionList.kt

```diff
@@ -1,31 +1,99 @@
 package io.github.droidkaigi.confsched2022.feature.sessions
 
-import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.LazyListState
 import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
 import androidx.compose.runtime.key
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import io.github.droidkaigi.confsched2022.model.TimetableItemId
 import io.github.droidkaigi.confsched2022.model.TimetableItemWithFavorite
 
 @Composable
 fun SessionList(
-    timetable: List<Pair<DurationTime?, TimetableItemWithFavorite>>,
+    timetable: List<Pair<DurationTime, TimetableItemWithFavorite>>,
     sessionsListListState: LazyListState,
+    onTimetableClick: (timetableItemId: TimetableItemId) -> Unit,
     modifier: Modifier = Modifier,
-    contentPadding: PaddingValues = PaddingValues(),
     content: @Composable (Pair<DurationTime?, TimetableItemWithFavorite>) -> Unit,
 ) {
-    LazyColumn(
-        modifier = modifier.fillMaxSize(),
-        state = sessionsListListState,
-        contentPadding = contentPadding,
+    Box(
+        modifier = modifier,
     ) {
-        itemsIndexed(timetable) { _, item ->
-            key(item.second.timetableItem.id.value) {
-                content(item)
+        val visibleItemsInfo = remember {
+            derivedStateOf {
+                sessionsListListState.layoutInfo.visibleItemsInfo
+            }
+        }
+        var currentDurationTime: DurationTime? = null
+        visibleItemsInfo.value.forEachIndexed { visibleItemIndex, visibleItemInfo ->
```

Comment:
Thanks for your great effort!
Probably this cause performance problem.
I think it is tough. But I came up with two approaches to solve this problem.

1. Use draw behind 
In this way, you can skip the composition and layout phase

```kotlin
        LazyColumn(
            modifier = Modifier.fillMaxSize().drawBehind {
                     ...
            }
```

3. Pass sessionsListListState and consume it in the function

This way, you can't skip the recompose, but you can narrow the recompose.

```kotlin
fun TimeLane(sessionsListListState, timetable) {
        val visibleItemsInfo = remember {
            derivedStateOf {
                sessionsListListState.layoutInfo.visibleItemsInfo
            }
        }
}

Reply:
I'm sorry for my late response.
Your advice helped me for fixing the code, thank you!
I adopted the second one in my code (I tried solving this by using drawBehind but I can't ...).

Reply:
Thanks!